pkg/cstorvolume/v1alpha1/build.go

```diff
@@ -0,0 +1,274 @@
+		)
+		return b
+	}
+	b.cstorvolume.object.Name = name
+	return b
+}
+
+// WithGenerateName sets the GenerateName field of
+// CStorVolume with provided value
+func (b *Builder) WithGenerateName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing cstorvolume generateName"),
+		)
+		return b
+	}
+
+	b.cstorvolume.object.GenerateName = name
+	return b
+}
+
+// WithNamespace sets the Namespace field of CStorVolume provided arguments
+func (b *Builder) WithNamespace(namespace string) *Builder {
+	if len(namespace) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing cstorvolume namespace"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Namespace = namespace
+	return b
+}
+
+// WithAnnotations merges existing annotations if any
+// with the ones that are provided here
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing annotations"),
+		)
+		return b
+	}
+
+	if b.cstorvolume.object.Labels == nil {
+		return b.WithAnnotationsNew(annotations)
+	}
+
+	for key, value := range annotations {
+		b.cstorvolume.object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithAnnotationsNew resets existing annotations if any with
+// ones that are provided here
+func (b *Builder) WithAnnotationsNew(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: no new annotations"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newannotations := map[string]string{}
+	for key, value := range annotations {
+		newannotations[key] = value
+	}
+
+	// override
+	b.cstorvolume.object.Annotations = newannotations
+	return b
+}
+
+// WithLabels merges existing labels if any
+// with the ones that are provided here
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing labels"),
+		)
+		return b
+	}
+
+	if b.cstorvolume.object.Labels == nil {
+		b.WithLabelsNew(labels)
+	}
+
+	for key, value := range labels {
+		b.cstorvolume.object.Labels[key] = value
+	}
+	return b
+}
+
+// WithLabelsNew resets existing labels if any with
+// ones that are provided here
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: no new labels"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newlbls := map[string]string{}
+	for key, value := range labels {
+		newlbls[key] = value
+	}
+
+	// override
+	b.cstorvolume.object.Labels = newlbls
+	return b
+}
+
+// WithTargetIP sets the TargetIP field of CStorVolume with provided arguments
```

Comment:
Please rephrase the comment.
How about below:
```go
// WithTargetIP sets target IP address of CStorVolume
```

Reply:
done